<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Runner</title>
    <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline';" />
    <script>
        function rejectPromise() {
          new Promise((_, reject) => {
            reject(new TypeError("invalid"))
          })
        }

        function handledError() {
          try {
            throw new Error('ALERT!')
          } catch (e) {
            window.Bugsnag.notify(e)
          }
        }

        function leaveCrumb() {
          window.Bugsnag.leaveBreadcrumb('missing auth token', { session: 'two-two' })
        }
    </script>
</head>
<body style="background: white;">
    <h1>haha electron go brrr</h1>
    <h2>actions</h2>
    <ul>
        <li><a href="#" id="main-process-crash" onclick="RunnerAPI.mainProcessCrash()">Crash in main process</a></li>
        <li><a href="#" id="main-process-uncaught-exception" onclick="RunnerAPI.mainProcessUncaughtException()">Uncaught exception in main process</a></li>
        <li><a href="#" id="main-process-unhandled-promise-rejection" onclick="RunnerAPI.mainProcessUnhandledPromiseRejection()">Unhandled promise rejection in main process</a></li>
        <li><a href="#" id="main-notify" onclick="RunnerAPI.mainProcessNotify()">Handled error in main process</a></li>
    </ul>
    <ul>
        <li><a href="#" id="renderer-uncaught-exception" onclick="foo()">Uncaught exception in renderer</a></li>
        <li><a href="#" id="renderer-unhandled-promise-rejection" onclick="rejectPromise()">Unhandled promise rejection in renderer</a></li>
        <li><a href="#" id="renderer-notify" onclick="handledError()">Handled error in renderer</a></li>
    </ul>
    <ul>
        <li><a href="#" id="custom-breadcrumb" onclick="leaveCrumb()">Leave a custom breadcrumb</a>
    </ul>
    <ul>
        <li><a href="#" id="main-process-start-session" onclick="RunnerAPI.mainProcessStartSession()">Start session in main process</a></li>
    </ul>
</body>
</html>
